modules/core/src/main/java/org/apache/synapse/aspects/newstatistics/StatisticsEntry.java

```diff
@@ -0,0 +1,510 @@
+/*
+ *   Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.apache.synapse.aspects.newstatistics;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.aspects.ComponentType;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+
+/**
+ * StatisticsEntry collects all the statistics logs related to a message flow. It is responsible
+ * for collecting statistics logs in correct hierarchy so that these logs can be directly fed
+ * into the statistic store as inputs
+ */
+public class StatisticsEntry {
+
+	private static final Log log = LogFactory.getLog(StatisticsEntry.class);
+
+	/**
+	 * ArrayList to hold all the statistic logs related to the message flow
+	 */
+	private final ArrayList<StatisticsLog> messageFlowLogs = new ArrayList<StatisticsLog>();
```

Comment:
Use proper abstract/super/template types as much as possible instead of sticking to low level implementations. For instance, you could have used "List" in place of "ArrayList" in your variable declaration.